core/markets/retrieval/common_types.hpp

```diff
@@ -0,0 +1,30 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef CPP_FILECOIN_CORE_MARKETS_RETRIEVAL_COMMON_TYPES_HPP
+#define CPP_FILECOIN_CORE_MARKETS_RETRIEVAL_COMMON_TYPES_HPP
+
+#include <libp2p/peer/peer_info.hpp>
+#include "primitives/address/address.hpp"
+
+namespace fc::markets::retrieval {
+  using libp2p::peer::PeerId;
+  using primitives::address::Address;
+
+  /**
+   * @struct Everything needed to make a deal with a miner
+   */
+  struct RetrievalPeer {
```

Comment:
We can use existing libp2p::peer::PeerInfo instead - it supports several addresses for each peer, maybe, it has sense (e.g. host supports tcp and udp and it will serve `/ip4/127.0.0.1/tcp/8080` and `/ip4/127.0.0.1/udp/8000`